### DISCUSSION TITLE
Flowise: Allow Langfuse traceId to be set or passed on with API response

### DISCUSSION THREAD

**Discussion by jpmin7** [langfuse/langfuse#1076] [Ideas]
### Describe the feature or potential improvement

The langfuse trace Id is needed to use the Langfuse toolset fully. Currently, we are blocked to iterate on flows. 

With the langfuse trace id we are allowed to a) implement user scoring on LLM responses and b) use the "Dataset" feature, which allows us to iterate and improve Flowise flows more efficiently. 

Potential solution: 
- allow users to set the traceId similar to setting the userId and the sessionId 
- pass on a flowise Id and add it to the api response 

### Contribute

- [ ] Yes, I can implement this and raise a PR

**Comment by marcklingen** [comment]
Agree that currently most platform features of Langfuse beyond tracing/monitoring are not usable without exposing the traceId.

I see three ways to solve this (ordered by the assumed ease of implementation):

1. use a flowise-internal id as the langfuse `traceId` - if there is an id which identifies a single flowise API invocation AND is returned with the response anyway
2. optionally set `traceId` via `overrideConfig.analytics.LangFuse` – user needs to generate unique ids
3. The Langfuse SDK exposes the id of a trace while it is created. Flowise could pass on this id. This also would not add latency as the id is created by the Langfuse SDK, but it would probably be more complex than options 1 and 2.
